.github/action_scripts/send_transactions/allow-spends-and-spend-transactions.js

```diff
@@ -417,6 +419,196 @@ const executeInvalidTransactionWorkflow = async (createTransactionFn, workflowNa
+            amount: allowSpend.amount,
+            fee: allowSpend.fee,
+            lastValidEpochProgress: allowSpend.lastValidEpochProgress,
+            dagL1Hash: response1.data.hash,
+            extendedDagL1Hash: response2.data.hash
+        };
+    } catch (error) {
+        logWorkflow.error('Error sending double spend transactions', error);
+        throw error;
+    }
+};
+
+const verifyDoubleSpendInL0 = async (address, hash, l0Url) => {
+    return await withRetry(
+        async () => {
+            logWorkflow.info(`Checking L0 for hash: ${hash} at address: ${address}`);
+            const { data: snapshot } = await axios.get(`${l0Url}/global-snapshots/latest/combined`);
+            
+            const activeAllowSpends = snapshot[1]?.activeAllowSpends?.['']?.[address] || [];
+            
+            if (activeAllowSpends.length === 0) {
+                logWorkflow.warning(`No active allow spends found for address ${address} in L0`);
+                throw new Error('No active allow spends found');
+            }
+            
+            logWorkflow.info(`Found ${activeAllowSpends.length} active allow spends for address ${address}`);
+            
+            const hasMatchingHash = await findMatchingHash(activeAllowSpends, hash);
+            
+            if (hasMatchingHash) {
+                logWorkflow.success(`Allow spend with hash ${hash} found in L0`);
+                return true;
+            } else {
+                logWorkflow.warning(`Allow spend with hash ${hash} not found in L0`);
+                return false;
+            }
+        },
+        { name: 'L0 double spend verification', maxRetries: 10, interval: 5000 }
+    );
+};
+
+const executeDoubleSpendWorkflow = async () => {
+    try {
+        logWorkflow.start('DoubleSpendAllowSpend');
+
+        const config = createConfig();
+        const urls = createNetworkConfig(config);
+        
+        logWorkflow.info(`Using DAG L1 URL: ${urls.dagL1Url}`);
+        logWorkflow.info(`Using extended DAG L1 URL: ${urls.extendedDagL1Url}`);
+        
+        const result = await createDoubleSpendTransaction(
+            PRIVATE_KEYS.key1, 
+            urls.dagL1Url, 
+            urls.extendedDagL1Url, 
+            urls.globalL0Url
+        );
+        
+        logWorkflow.info('Double spend transactions sent successfully, waiting for L0 processing...');
+        
+        await sleep(CONSTANTS.SNAPSHOT_WAIT_TIME_MS);
+        
+        const isDagL1HashAccepted = await verifyDoubleSpendInL0(
+            result.address, 
+            result.dagL1Hash, 
+            urls.globalL0Url
+        );
+        
+        const isExtendedDagL1HashAccepted = await verifyDoubleSpendInL0(
+            result.address, 
+            result.extendedDagL1Hash, 
+            urls.globalL0Url
+        );
+        
+        const exactlyOneAccepted = (isDagL1HashAccepted && !isExtendedDagL1HashAccepted) || (!isDagL1HashAccepted && isExtendedDagL1HashAccepted);
```

Comment:
As long as we're returning the number of allow spends in the state, we should also be checking that to verify it matches what we expect. In case this test is run with other tests, at the beginning we should check the number and verify the count is incremented by 1.

Reply:
> As long as we're returning the number of allow spends in the state

We're not returning the number of allow spends in the state

> we should also be checking that to verify it matches what we expect

And this is exactly what this code does 🤷‍♂️ 

Why we should care about other allow spends if there are any? What we need to verify is that only one of both allow spend has been accepted.

Reply:
I've changed to verifying the number of allow spends in the state